### PR TITLE
Makes abductor baton child of stun baton

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request
- Main change is the abductor baton's stun mode first uses the security baton's attack proc, then its own stun proc (i.e. 14 second paralyze instead of regular baton's 5 second knockdown). The only functional difference this means is that the stunned person will jitter (sprite shakes), but it removes copypasted code and allows for the abductor baton to be consistent with the regular one.
- Adds vars to stun baton for stun delay, stamina damage amount, confusion amount, off/on sound and stun sound
- Adds a different suicide if a stun baton is off or out of charge
- Adds vars to abductor baton for sleeptime, cufftime and whether the cell can be removed
- Sets abductor's baton stun force to 14 seconds, and its stun delay, cooldown, stamina damage, confusion and toggle sound to nothing (how it was before)
- Abductor baton has an infinite cell, can't be removed (how it worked before)
- Uses defines instead of magic numbers
- Makes shield check its own proc to reduce copypaste code
- Removes a lot of copypasted/duplicate code
- Renames some baton vars to make it clearer what they do or what they measure in
- Fixes a couple of typos
- Updates abductor baton typepaths in files
- Makes AbductorCheck(user) an item proc rather than an item/abductor proc to make it compatible with the refactored baton
- Makes it so you need to be an abductor to change baton modes

## Why It's Good For The Game

Exposes vars for admins and balance. Reduces copypasted code. Gives the abductor baton more of the functionality of the regular baton. 

## Changelog
:cl:
refactor: The abductor baton is now a child of the regular stun baton. There shouldn't be a functional difference.
tweak: You need to be an abductor to change baton modes. Seemed like an oversight
spellcheck: Fixes some typos in the abductor baton's messages
admin: Both regular stun baton and abductor baton have had a lot of vars exposed pertaining to their functionality so that could come in handy I guess
/:cl:

### CODE REVIEW

**Comment by bobbahbrown** [comment]
```
_maps/shuttles/emergency_zeta.dmm:70:error: undefined type: /obj/item/abductor/baton
_maps/shuttles/emergency_zeta.dmm:71:error: invalid variable name: desc
_maps/shuttles/emergency_zeta.dmm:74:error: unknown type```

map edit necessary

**Comment by gitbirb** [comment]
i thought abductor baton was like this so that abductors are actually able to abduct people and not get memed on, and so it can combo into sleep instantly.

**Comment by Jalleo** [comment]
Good idea. I think this is a good step. The stun baton itself may need to be setup to use components at some point.

**Comment by cacogen** [comment]
Functionality isn't any different, download it and see for yourself.
-- 
Sent from my phone

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 1]
```diff
@@ -430,18 +430,39 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
 	force = 7
+
 	w_class = WEIGHT_CLASS_NORMAL
+	slot_flags = ITEM_SLOT_BELT
+
 	actions_types = list(/datum/action/item_action/toggle_mode)
 
-/obj/item/abductor/baton/proc/toggle(mob/living/user=usr)
+	preload_cell_type = /obj/item/stock_parts/cell/infinite //Any sufficiently advanced technology is indistinguishable from magic
+	var/can_remove_cell = FALSE
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: New vars at the bottom

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by kriskog** [comment]
Merge conflict

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 2]
```diff
@@ -436,22 +436,45 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
-	slot_flags = ITEM_SLOT_BELT
+
 	force = 7
+
 	w_class = WEIGHT_CLASS_NORMAL
+	slot_flags = ITEM_SLOT_BELT
+
 	actions_types = list(/datum/action/item_action/toggle_mode)
 
-/obj/item/absductor/baton/ComponentInitialize()
```
Comment (cacogen) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: >absductor

i wonder what this is causing

**Comment by cacogen** [comment]
@kriskog
fix'd

**Comment by optimumtact** [comment]
conflicts

File: code/game/objects/items/stunbaton.dm [hunk 3]
```diff
@@ -1,34 +1,50 @@
...
+
+	w_class = WEIGHT_CLASS_NORMAL
+	slot_flags = ITEM_SLOT_BELT
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 50, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 80)
 
-	var/cooldown_check = 0
+	throwforce = 7
+	var/throw_stun_chance = 35
+
+	var/attack_cooldown_check = (0 SECONDS)
```
Comment (AnturK) [inline code/game/objects/items/stunbaton.dm]: What's with all the ()'s, they're not necessary.

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 4]
```diff
@@ -497,45 +531,38 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
-	L.apply_effect(EFFECT_STUTTER, 7)
-	SEND_SIGNAL(L, COMSIG_LIVING_MINOR_SHOCK)
+/obj/item/melee/baton/abductor/attack_self(mob/living/user)
+	toggle(user)
 
-	L.visible_message("<span class='danger'>[user] has stunned [L] with [src]!</span>", \
-							"<span class='userdanger'>[user] has stunned you with [src]!</span>")
-	playsound(src, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
+/obj/item/melee/baton/abductor/attackby(obj/item/W, mob/user, params)
+	if(can_remove_cell)
```
Comment (AnturK) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: That's too wide, extract battery removal into separate proc and check there.

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks alright just one minor change.

File: code/game/objects/items/stunbaton.dm (hunk 3 above)
Comment (cacogen) [reply code/game/objects/items/stunbaton.dm]: following wrong precedent

**Pull request merged**